applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.h

```diff
@@ -0,0 +1,394 @@
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template< class TElementData >
+class DVMSDEMCoupled : public DVMS<TElementData>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of DVMSDEMCoupled
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(DVMSDEMCoupled);
+
+    /// Node type (default is: Node<3>)
+    typedef Node<3> NodeType;
+
+    /// Geometry type (using with given NodeType)
+    typedef Geometry<NodeType> GeometryType;
+
+    /// Definition of nodes container type, redefined from GeometryType
+    typedef Geometry<NodeType>::PointsArrayType NodesArrayType;
+
+    /// Vector type for local contributions to the linear system
+    typedef Vector VectorType;
+
+    /// Matrix type for local contributions to the linear system
+    typedef Matrix MatrixType;
+
+    typedef std::size_t IndexType;
+
+    typedef std::size_t SizeType;
+
+    typedef std::vector<std::size_t> EquationIdVectorType;
+
+    typedef std::vector< Dof<double>::Pointer > DofsVectorType;
+
+    typedef PointerVectorSet<Dof<double>, IndexedObject> DofsArrayType;
+
+    /// Type for shape function values container
+    typedef Kratos::Vector ShapeFunctionsType;
+
+    /// Type for a matrix containing the shape function gradients
+    typedef Kratos::Matrix ShapeFunctionDerivativesType;
+
+    /// Type for an array of shape function gradient matrices
+    typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionDerivativesArrayType;
+
+    constexpr static unsigned int Dim = DVMS<TElementData>::Dim;
+    constexpr static unsigned int NumNodes = DVMS<TElementData>::NumNodes;
+    constexpr static unsigned int BlockSize = DVMS<TElementData>::BlockSize;
+    constexpr static unsigned int LocalSize = DVMS<TElementData>::LocalSize;
+    constexpr static unsigned int StrainSize = DVMS<TElementData>::StrainSize;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    //Constructors.
+
+    /// Default constuctor.
+    /**
+     * @param NewId Index number of the new element (optional)
+     */
+    DVMSDEMCoupled(IndexType NewId = 0);
+
+    /// Constructor using an array of nodes.
+    /**
+     * @param NewId Index of the new element
+     * @param ThisNodes An array containing the nodes of the new element
+     */
+    DVMSDEMCoupled(IndexType NewId, const NodesArrayType& ThisNodes);
+
+    /// Constructor using a geometry object.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     */
+    DVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    /// Constuctor using geometry and properties.
+    /**
+     * @param NewId Index of the new element
+     * @param pGeometry Pointer to a geometry object
+     * @param pProperties Pointer to the element's properties
+     */
+    DVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties);
+
+    /// Destructor.
+    ~DVMSDEMCoupled() override;
```

Comment:
Destructors shouldn't be `override` specified [(see here)](https://github.com/isocpp/CppCoreGuidelines/issues/721).